doc/doxygen/src/Examples.dox

```diff
@@ -277,9 +277,14 @@
  *
  * # Whenever we make new data, we always tell the manager first,
  * # This allows it to create a placeholder version or similar.
+ * # We must provide the manager with as much information as possible
+ * # up-front, to help it prepare the backend and determine the most
+ * # appropriate response to our subsequent `resolve` and `register`
+ * # calls, or error early if publishing is likely to fail.
+ * preflight_hint = TextFileSpecification.create()
+ * preflight_hint.markupTrait().setMarkupType("plain")
```

Comment:
[Consider] Perhaps another trait here would be good for the example, to convey that you should provide all the data you have, (if we do want to convey that,) rather than just the data you "know" your specific manager cares about. (Which i guess you could have worked out from trial and error.)

Reply:
The "owning" part is the way you decide, which feels to remove the ambiguity.

Also we need to make sure we set the markupTrait for `register` too. It might be worth just assembling the final data at the beginning, vs making these a separate traits data? (Ie move `file_spec` up)

Reply:
Moved `file_spec` up in daa8a1ac